stdlib/src/collections/list.mojo

```diff
@@ -239,6 +239,73 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             earlier_idx -= 1
             later_idx -= 1
 
+    @always_inline
+    fn __mul(inout self, x: Int):
+        """Appends the original elements of this list x-1 times.
+
+        ```mojo
+        var a = List[Int](1, 2)
+        a.__mul(2) # a = [1, 2, 1, 2]
+        ```
+
+        Args:
+            x: The multiplier number.
+        """
+        if x == 0:
+            self.clear()
+        var orig = List(self)
+        for i in range(x - 1):
+            self.extend(orig)
+
+    @always_inline("nodebug")
+    fn __mul__(self, x: Int) -> Self:
+        """Multiplies the list by x and returns a new list.
+
+        Args:
+            x: The multiplier number.
+
+        Returns:
+            The new list.
+        """
+        # avoid the copy since it would be cleared immediately anyways
+        if x == 0:
+            return Self()
+        var result = List(self)
+        result.__mul(x)
+        return result^
+
+    @always_inline("nodebug")
+    fn __imul__(inout self, x: Int):
+        """Multiplies the list by x in place.
+
+        Args:
+            x: The multiplier number.
+        """
+        self.__mul(x)
+
+    @always_inline("nodebug")
+    fn __add__(self, owned other: Self) -> Self:
```

Comment:
I think this should return a new list. Without conditional conformance, I guess we can't achieve the desired semantics as it requires `Copy`.